### ISSUE TITLE
[a11y]: No announcement of Inline Notification type by screen reader

### ISSUE THREAD

**Issue by kshitiz-ibm** [carbon-design-system/carbon#20532]
### Package

@carbon/react

### Package version

1.80.1

### React version

17.0.2

### Automated testing tool and ruleset

WCAG Mapping: 2.4.6 Headings and Labels, Test tool: JAWS_Firefox

### Assistive technology

JAWS, Mac Voice Over

### Description

The Carbon Inline Notification close icon label doesn't specify what type of notification it is (error, success, information, etc), so when read by a  screen reader it just reads the notification but the user has no idea on what type of notification it is. 

This issue was discovered when I was trying to fic the issue here - https://github.ibm.com/devx-app-services/planning/issues/23045
Requesting that the information about type of notification be added to the JAWS label.y.

### WCAG 2.1 Violation

WCAG Mapping: 2.4.6 Headings and Labels

### Reproduction/example

-

### Steps to reproduce

Try any screen reader and read the notification element, notice how the type of notification is not read.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by heloiselui** [comment]
I checked with the accessibility team, and it’s indeed an issue we need to fix.
